drivers: modem: hl7800: fix update RAT

rerickson1 opened the pull request:
The default behavior of the +KSRAT= command has
changed to not reboot the HL7800.
Adjust the command so the reboot takes place properly.

jukkar commented on drivers/modem/hl7800.c:
```diff
@@ -347,6 +348,14 @@ static const char TIME_STRING_FORMAT[] = "\"yy/MM/dd,hh:mm:ss?zz\"";
 	} while (0)
 
+#define SEND_AT_CMD_ARGS_EXPECT_OK(c, t, n)                                    \
+	do {                                                                   \
+		ret = send_at_cmd(NULL, (c), (t), (n), false);                 \
+		if (ret < 0) {                                                 \
+			LOG_ERR("%s result:%d", (c), ret);                     \
+		}                                                              \
+	} while (0)
```
I wonder if we could do this slightly differently like this
```
#define SEND_AT_CMD_ARGS_EXPECT_OK(c, t, n) \
({                                                                      \
	ret = send_at_cmd(NULL, (c), (t), (n), false);                 \
	if (ret < 0) {                                                 \
		LOG_ERR("%s result:%d", (c), ret);                     \
	}                                                                 \
        ret;                                                             \
})
```
This way we could do
```
ret = SEND_AT_CMD_ARGS_EXPECT_OK("AT+KSRAT=0", MDM_CMD_KSRAT_TIMEOUT, 0);
if (ret < 0) {
	goto error;
}
```
which would not "hide" the ret value. If one reads the patch, it is not obvious where the value of ret comes from because it is hidden inside the macro.

rerickson1 committed:
drivers: modem: hl7800: fix update RAT

The default behavior of the +KSRAT= command has
changed to not reboot the HL7800.
Adjust the command so the reboot takes place properly.

Signed-off-by: Ryan Erickson <ryan.erickson@lairdconnect.com>
